jupyter_server/base/handlers.py

```diff
@@ -184,7 +184,16 @@ def login_available(self):
         return bool(self.login_handler.get_login_available(self.settings))
 
 
-class JupyterHandler(AuthenticatedHandler):
+class AuthorizedHandlerMixin:
+    """A mixin class for Tornado request handlers that checks whether
+    the current user is authorized to execute the current action.
+    """
+    def user_is_authorized(self, user, action, resource):
+        """Check is `user` is authorized to do `action` on given `resource`."""
```

Comment:
Great feedback. Thanks, Kevin. 

I originally chose the `user_is_authorized` name to work inline with tornado's authentication mechanism. The `@authorized` wrapper uses the authenticated *user* from `.get_current_user()` to get the subject for authorization.

I like the idea of dropping "user" from the function signature. I just want to make it as clear as possible that the first argument passed in by `@authorized` will always be the returned value of `.get_current_user()`.

Reply:
Just a note that "user" is probably a lot easier to understand for people who doesn't have English as their first language :)

Reply:
Good point.  Perhaps we could change the method name to `is_authorized()` since the `user` parameter already indicates the subject (sorry, pun intended :smile:) of the authorization.